community/planning/cylinder_jwt.md

```diff
@@ -0,0 +1,227 @@
+# Cylinder JWT
+<!--
+  Copyright 2018-2020 Cargill Incorporated
+  Licensed under Creative Commons Attribution 4.0 International License
+  https://creativecommons.org/licenses/by/4.0/
+-->
+
+## Summary
+[summary]: #summary
+
+This proposal adds a JSON Web Token (JWT) module to the Cylinder library.
+
+## Motivation
+[motivation]: #motivation
+
+Implementing a JWT feature in Cylinder allows for the use of its own signer
+implementations to provide signed tokens. This provides Cylinder users to make
+use of the same keys as an identity for use in both HTTP authentication tokens
+(via JWT) as well as the current comment use case of transaction signers.
+
+Existing JWT libraries do not allow for the flexibility to use custom signers.
+
+##Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Cylinder JWT provides a set of API's to both generate a JWT-like token and
+cryptographically validate its contents.
+
+The header will follow the standard minimally consist of a type (`"typ"`) of
+`"cylinder+jwt"`, and an algorithm (`"alg"`) value that depends on the Signer
+implementation.
+
+The use of the term JWT-like is necessary as the algorithms currently provided
+by Cylinder are not part of the standard set.  The resulting token will still
+follow the format of
+
+```
+<base64 encoded bytes of the utf8 string of the header json>.\
+<base64 encoded bytes of the utf8 string of the claims json>.\
+<base64 signature>
+```
+
+While this format will still be parseable by other JWT libraries, most likely
+the signing algorithm specified in the header will not be understood.
+
+The initial implementation will only provide the ability to create flat JSON
+objects, both for the header and for the claims.  Complex, nested JSON objects
+are beyond the scope of this initial design.
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### JWT API
+[jwt-api]: #jwt-api
+
+This module will be guarded by the feature "jwt".
+
+```rust
+mod cylinder::jwt;
+```
+
+The module will provide a struct for building the JWT string value.  This
+builder will optionally accept a set of header values and set of claims.  At
+build time, it will sign the token.  Note, in the following API, implementations
+have been omitted.
+
+```rust
+/// Builder for constructing the JWT string that would be included in request
+/// headers
+struct JsonWebTokenBuilder {
+    ...
+}
+
+impl JsonWebTokenBuilder {
+    /// Constructs a new instance of the builder.
+    pub fn new() -> Self {
+        …
```

Comment:
Change this to . . .  like the other instances.

Reply:
fixed